includes/class-wc-meta-data.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * Wraps an array (meta data for now) and tells if there was any changes.
+ *
+ * The main idea behind this class is to avoid doing unneeded
+ * SQL updates if nothing changed.
+ *
+ * @version 3.0.x
+ * @package WooCommerce
+ * @category Class
+ * @author crodas
+ */
+class WC_Meta_Data {
+	public $id;
```

Comment:
Rather than defining these publicly, how about implementing a magic getter and setter which can also track the changes prior to setting the values?